sdk/database/newdbplugin/grpc_server.go

```diff
@@ -0,0 +1,190 @@
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var _ proto.DatabaseServer = gRPCServer{}
+
+type gRPCServer struct {
+	impl Database
+}
+
+// Initialize the database plugin
+func (g gRPCServer) Initialize(ctx context.Context, request *proto.InitializeRequest) (*proto.InitializeResponse, error) {
+	rawConfig := structToMap(request.ConfigData)
+
+	dbReq := InitializeRequest{
+		Config:           rawConfig,
+		VerifyConnection: request.VerifyConnection,
+	}
+
+	dbResp, err := g.impl.Initialize(ctx, dbReq)
+	if err != nil {
+		return &proto.InitializeResponse{}, status.Errorf(codes.Internal, "failed to initialize: %s", err)
+	}
+
+	newConfig, err := mapToStruct(dbResp.Config)
+	if err != nil {
+		return &proto.InitializeResponse{}, status.Errorf(codes.Internal, "failed to marshal new config to JSON: %s", err)
+	}
+
+	resp := &proto.InitializeResponse{
+		ConfigData: newConfig,
+	}
+
+	return resp, nil
+}
+
+func (g gRPCServer) NewUser(ctx context.Context, req *proto.NewUserRequest) (*proto.NewUserResponse, error) {
+	if req.GetUsernameConfig() == nil {
+		return &proto.NewUserResponse{}, status.Errorf(codes.InvalidArgument, "missing username config")
+	}
+
+	var expiration time.Time
+
+	if req.GetExpiration() != nil {
+		exp, err := ptypes.Timestamp(req.GetExpiration())
+		if err != nil {
+			return &proto.NewUserResponse{}, status.Errorf(codes.InvalidArgument, "unable to parse expiration date: %s", err)
+		}
+		expiration = exp
+	}
+
+	dbReq := NewUserRequest{
+		UsernameConfig: UsernameMetadata{
+			DisplayName: req.GetUsernameConfig().GetDisplayName(),
+			RoleName:    req.GetUsernameConfig().GetRoleName(),
+		},
+		Password:           req.GetPassword(),
+		Expiration:         expiration,
+		Statements:         getStatementsFromProto(req.GetStatements()),
+		RollbackStatements: getStatementsFromProto(req.GetRollbackStatements()),
+	}
+
+	dbResp, err := g.impl.NewUser(ctx, dbReq)
+	if err != nil {
+		return &proto.NewUserResponse{}, status.Errorf(codes.Internal, "unable to create new user: %s", err)
+	}
+
+	resp := &proto.NewUserResponse{
+		Username: dbResp.Username,
+	}
+	return resp, nil
+}
+
+func (g gRPCServer) UpdateUser(ctx context.Context, req *proto.UpdateUserRequest) (*proto.UpdateUserResponse, error) {
+	if req.GetUsername() == "" {
+		return &proto.UpdateUserResponse{}, status.Errorf(codes.InvalidArgument, "no username provided")
+	}
+
+	dbReq, err := getUpdateUserRequest(req)
+	if err != nil {
+		return &proto.UpdateUserResponse{}, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	_, err = g.impl.UpdateUser(ctx, dbReq)
+	if err != nil {
+		return &proto.UpdateUserResponse{}, status.Errorf(codes.Internal, "unable to update user: %s", err)
+	}
+	return &proto.UpdateUserResponse{}, nil
+}
+
+func getUpdateUserRequest(req *proto.UpdateUserRequest) (UpdateUserRequest, error) {
+	var password *ChangePassword
+	if req.GetPassword() != nil && req.GetPassword().GetNewPassword() != "" {
```

Comment:
`GetNewPassword()` has a built-in nil check on the receiver so you shouldn't need ` req.GetPassword() != nil` to protect against potential panic, but it's good to be explicit.